packages/across-monitor/src/AcrossMonitor.ts

```diff
@@ -0,0 +1,151 @@
+  ) {
+    this.relayEventProcessor = new RelayEventProcessor();
+  }
+
+  async update(): Promise<void> {
+    // Update l1Client for bridge pool discovery.
+    await this.l1Client.update();
+    this.bridgePools = this.l1Client.bridgePools;
+    this.relayEventProcessor.bridgePools = this.l1Client.bridgePools;
+
+    // In serverless mode (pollingDelay === 0) use block range from environment (or just the latest block if not provided) to fetch for latest events.
+    // Else, if running in loop mode (pollingDelay != 0), start with the latest block and on next loops continue from where the last one ended.
+    const latestL1BlockNumber = await this.l1Client.l1Web3.eth.getBlockNumber();
+    if (this.monitorConfig.pollingDelay === 0) {
+      this.startingBlock =
+        this.monitorConfig.startingBlock !== undefined ? this.monitorConfig.startingBlock : latestL1BlockNumber;
+      this.endingBlock =
+        this.monitorConfig.endingBlock !== undefined ? this.monitorConfig.endingBlock : latestL1BlockNumber;
+    } else {
+      this.startingBlock = this.endingBlock ? this.endingBlock + 1 : latestL1BlockNumber;
+      this.endingBlock = latestL1BlockNumber;
+    }
+    // Starting block should not be after the ending block (this could happen on short polling period or misconfiguration).
+    this.startingBlock = Math.min(this.startingBlock, this.endingBlock);
+
+    await this.relayEventProcessor.update(this.endingBlock);
+  }
+
+  async checkUtilization(): Promise<void> {
+    this.logger.debug({ at: "AcrossMonitor#Utilization", message: "Checking for pool utilization ratio" });
+
+    // Collect utilization and other properties for all bridge pools.
+    const bridgePools = await Promise.all(
+      Object.keys(this.bridgePools).map(async (l1TokenAddress) => {
+        const utilization = await this.bridgePools[l1TokenAddress].contract.methods
+          .liquidityUtilizationCurrent()
+          .call();
+        return {
+          address: this.bridgePools[l1TokenAddress].contract.options.address,
+          chainId: this.monitorConfig.bridgeAdminChainId,
+          poolCollateralSymbol: this.bridgePools[l1TokenAddress].poolCollateralSymbol,
+          utilization: utilization,
+        };
+      })
+    );
+
+    // Send notification if pool utilization is above configured threshold.
+    for (const bridgePool of bridgePools) {
+      if (
+        toBN(bridgePool.utilization).gt(
+          toBN(this.monitorConfig.utilizationThreshold)
+            .mul(toBN(toWei("1")))
+            .div(toBN(100))
+        )
+      ) {
+        this.logger.warn({
+          at: "UtilizationMonitor",
+          message: "Across bridge pool utilization warning⚠",
+          mrkdwn:
+            bridgePool.poolCollateralSymbol +
+            " bridge pool at " +
+            createEtherscanLinkMarkdown(bridgePool.address, bridgePool.chainId) +
+            " on " +
+            PublicNetworks[bridgePool.chainId]?.name +
```

Comment:
this part of the log is likely not very useful as the pools are always on mainnet but it's ok to leave for now as this'll be forwards compatible when we move to multiple chains ;)